sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1107,6 +1112,9 @@ class Analyzer(override val catalogManager: CatalogManager)
       if (isStreaming && tmpView.nonEmpty && !tmpView.get.isStreaming) {
         throw QueryCompilationErrors.readNonStreamingTempViewError(identifier.quoted)
       }
+      if (isTimeTravel && tmpView.nonEmpty) {
+        throw QueryCompilationErrors.viewNotSupportTimeTravelError(identifier)
```

Comment:
Not quite related to this PR, but permanent view already fails with time travel, and temp view should be consistent.